[Workers] Version previews behind Access (#17239)
cloudflare/cloudflare-docs: src/content/docs/workers/configuration/previews.mdx

Preview URLs allow you to preview new versions of your Worker without deploying it to production.

> Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `versionPrefix.workerName.subdomainName.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.

Reviewer: Generally, you want to distinguish between variables and any fixed content in concatenated strings like this.

Also, I thought it was `<VERSION_PREFIX>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev` not `<VERSION_PREFIX>.<WORKER_NAME>.<SUBDOMAIN>.workers.dev`??

```suggestion
Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `<VERSION_PREFIX>.<WORKER_NAME>.<SUBDOMAIN>.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.
```

Author: You're right on format. Fixing

> Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `versionPrefix.workerName.subdomainName.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.

Author: Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `<VERSION_PREFIX>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.

> Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `versionPrefix.workerName.subdomainName.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.

Author: ```suggestion
Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `<VERSION_PREFIX>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.
```

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/workers) and select your project.
2. Head to the "Deployments" tab find the version you would like to view.

## Manage access to Preview URLs

By default, preview URLs are enabled and available publicly. You can use [Cloudflare Access](/cloudflare-one/policies/access/) to require visitors to authenticate before accessing preview URLs. You can limit access to yourself, your teammates, your organization, or anyone else you specify in your [access policy](/cloudflare-one/policies/access).

To limit your preview URLs to authorized emails only:

1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps)
2. Select your account
3. Add an application
4. Select "Self Hosted"

> 5. Name your application (for example, My-Worker) and add your `workers.dev` subdomain as the "Application domain".

Reviewer: ```suggestion
5. Name your application (for example, "my-worker") and add your `workers.dev` subdomain as the "Application domain".
```

Author: ugh AI linter. should have checked more carefully

> 5. Name your application (for example, "my-worker") and add your `workers.dev` subdomain as the "Application domain".

Author: ```suggestion
5. Name your application (for example, "my-worker") and add your `workers.dev` subdomain as the **Application domain**.
```

> For example, if you want to secure preview URLs for a Worker running on `my-worker.my-subdomain.workers.dev`.

Reviewer 2: This reads a bit incomplete. It seems like it should say "if you want to secure preview URLs, do x" but that latter part is missing.

> - Subdomain: `*.my-worker`

Reviewer: Again, I thought it was `*-my-worker`? Did this change?

> - Subdomain: `*.my-worker`

Author: ```suggestion
- Subdomain: `*-my-worker`
```

- Domain: `my-subdomain.workers.dev`

:::note
You must press enter after you input your Application domain for it to save. You will see a "Zone is not associated with the current account" warning that you may ignore.
:::

> 6. Go to the next page

Reviewer 2: ```suggestion
6. Go to the next page.
```

> 7. Add a name for your access policy (for example, Authorized preview emails for my-Worker)

Reviewer: ```suggestion
7. Add a name for your access policy (for example, "Allow employees access to preview URLs for my-worker")
```

> 7. Add a name for your access policy (for example, Authorized preview emails for my-Worker)

Reviewer 2: ```suggestion
7. Add a name for your access policy (for example, Authorized preview emails for my-Worker).
```

> 7. Add a name for your access policy (for example, "Allow employees access to preview URLs for my-worker")

Author: ```suggestion
7. Add a name for your access policy (for example, "Allow employees access to preview URLs for my-worker").
```

> 8. In the "Configure rules" section create a new rule with the "Emails" selector. Enter the emails you want to authorize to access the version preview. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.

Reviewer: ```suggestion
8. In the "Configure rules" section create a new rule with the "Emails" selector, or any other attributes which you wish to gate access to previews with. Enter the emails you want to authorize. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```

> 8. In the "Configure rules" section create a new rule with the "Emails" selector. Enter the emails you want to authorize to access the version preview. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.

Reviewer 2: ```suggestion
8. In the **Configure rules** section, create a new rule with the **Emails** selector. 
9. Enter the emails you want to authorize to access the version preview. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```

> 9. Go to the next page

Reviewer 2: ```suggestion
10. Go to the next page.
```

> 10. Add application

Reviewer 2: ```suggestion
11. Add application.
```

> 8. In the "Configure rules" section create a new rule with the "Emails" selector, or any other attributes which you wish to gate access to previews with. Enter the emails you want to authorize. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.

Author: ```suggestion
8. In the **Configure rules** section create a new rule with the **Emails** selector, or any other attributes which you wish to gate access to previews with. 
9. Enter the emails you want to authorize. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```

> 9. Go to the next page

Author: ```suggestion
9. Go to the next page.
```

> 9. Go to the next page.

Author: ```suggestion
10. Go to the next page.
```

> 10. Add application

Author: ```suggestion
10. Add application.
```

> 10. Add application.

Author: ```suggestion
11. Add application.
```
